File: src/characters/recluse.ml
open! Core
open! Import

include Character.Make (struct
    type t =
      { kind : Kind.t
      ; alignment : Alignment.t
      }

    let id = "recluse"
    let name = "Recluse"
    let t = { kind = Kind.Outsider; alignment = Alignment.Good }
    let kind { kind; _ } = kind
    let alignment { alignment; _ } = alignment
  end)

let night_action ~player_id:_ ~night:_ = None
let day_action ~player_id:_ = None
let on_setup ~player_id:_ = None
let on_nominated ~player_id:_ ~nominator:_ = None
let on_executed ~player_id:_ = None
let on_night_kill ~player_id:_ = None

File: src/characters/recluse.rs
//! Recluse (Outsider): you might register as evil, and as a Minion or Demon,
//! even though you are good.

use crate::ability::Ability;
use crate::event::Registration;
use crate::role::{Alignment, CharacterInfo, Kind};

character!(
    Recluse,
    "recluse",
    "Recluse",
    Kind::Outsider,
    Alignment::Good
);

impl Ability for Recluse {
    fn info(&self) -> CharacterInfo {
        Self::INFO
    }

    fn misregistration(&self) -> Option<Registration> {
        Some(Registration::recluse())
    }
}
